content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx

```diff
@@ -0,0 +1,238 @@
+tocDepth: 2
+preview: true
+---
+
+<TopBlock>
+
+<Admonition type="info">
+
+Embedded documents are available in Preview in versions 3.10.0 and later.
+
+</Admonition>
+
+<Admonition type="warning">
+
+Embedded documents are currently only available on MongoDB.
+
+</Admonition>
+
+Embedded documents introduce a new way of linking records in Prisma. There are two ways to establish a link between two records:
+
+- **Creating a relation**, by using a foreign key to link from one record to another record.
+- **Embedding a record**, by storing a record within another record.
+
+Prisma has great [support for relations](/concepts/components/prisma-schema/relations), but support for embedded data has previously been limited to the following options:
+
+- Using [scalar lists](/concepts/components/prisma-client/working-with-fields/working-with-scalar-lists-arrays) to store multiple values in a single field. Only lists of scalar fields are supported.
+- Using [`Json` fields](/concepts/components/prisma-client/working-with-fields/working-with-json-fields) to store JSON data within a single field. `Json` fields are untyped.
+
+Embedded documents introduce a new way to embed data:
+
+- **Embedded documents** allow you to define new object types. Embedded documents are structured and typed, similar to how models are structured and typed today. By understanding the contents of the embedded data, we can offer a richer set of operations you can perform on that data.
+
+As an example of the syntax, take the following schema:
+
+```prisma file=schema.prisma
+model Product {
+id     String  @id
+name   String
+photos Photo[]
+}
+
+type Photo {
+height Int
+width  Int
+url    String
+}
+```
+
+In this case, the `Product` model has an embedded `Photo[]` type.
+
+</TopBlock>
+
+## Using embedded documents
+
+This section explains how embedded documents can be created and updated. We’ll use this schema for the examples that follow:
+
+```prisma file=schema.prisma
+model Product {
+  id     String  @id
+  name   String
+  price  Decimal
+  colors Color[]
+  sizes  Size[]
+  photos Photo[]
+}
+
+model Order {
+  id              String   @id
+  product         Product
+  color           Color
+  size            Size
+  shippingAddress Address
+  billingAddress  Address?
+}
+
+enum Color {
+  Red
+  Green
+  Blue
+}
+
+enum Size {
+	Small
+	Medium
+	Large
+  XLarge
+}
+
+type Photo {
+  height Int
+  width  Int
+  url    String
+}
+
+type Address {
+  street String
+  city   String
+  zip    String
+}
+```
+
+In this schema, the `Product` model has a `Photo[]` embedded document, and the `Order` model has two embedded `Address` types. The `shippingAddress` is required, but the `billingAddress` is optional.
+
+### Creating embedded documents
+
+The value of an embedded document can be set within a `create` or `createMany` method using the new `set` method. For example, you can use `set` within `create` to model the case where an `order` contains one `shipping address`:
```

Comment:
IMO, what operations you can perform under what circumstances are the meat of this page:

Right now it feels like these are hidden within the paragraph.

I like what we do with the include & select: 
![CleanShot 2022-02-15 at 12 15 54](https://user-images.githubusercontent.com/170299/154123844-363b5a5e-35f0-402b-b622-b6bc63dad5fc.png)

There may be some redundancy with the Client API Reference though, but I guess my mental model is this page explains how you operate on embedded documents, whereas the reference is more of a handbook when you already know.

Reply:
(Reminder for me: screenshot is from https://www.prisma.io/docs/concepts/components/prisma-client/select-fields )

Reply:
The rewritten version should be closer to the Select Fields page (though there are no links to API reference docs as they don't exist yet...)